lib/internal/modules/cjs/loader.js

```diff
@@ -837,7 +850,8 @@ Module._load = function(request, parent, isMain) {
 };
 
 Module._resolveFilename = function(request, parent, isMain, options) {
-  if (NativeModule.canBeRequiredByUsers(request)) {
+  if (StringPrototypeStartsWith(request, 'node:') ||
+      NativeModule.canBeRequiredByUsers(request)) {
```

Comment:
This bifurcation will have implications for mocking, but I can appreciate the benefit too.

I'd still prefer a more convergent path here eg, to return `node:fs` for both `node:fs` and `fs` inputs, and then to still populate a `cache['fs']` entry in the `node:fs` case as the same object for backwards compatibility. Such a change would hopefully be mostly compatible but would probably need to be a separate major nontheless. Worth thinking about at least.

Reply:
Existing mocking tools (such as **Jest**) provide their own `require(…)` implementation, since they need to be able to bypass the mocked module using <code>[jest.requireActual(…)](https://jestjs.io/docs/en/bypassing-module-mocks)</code>.